mobile/lib/screens/main/home/home_screen.dart

```diff
@@ -124,40 +129,67 @@ class HomeScreen extends StatelessWidget {
   }
 
   Widget _buildContentRow() {
-    return const ContentRow(
-      imageUrl: 'https://media.discordapp.net/attachments/763282727826227202/877648551121915984/unknown.png',
-      title: 'We are who',
-      subtitle: 'Olivia Whodrigo | ONLY WHO',
+    return BlocProvider<RecentlyPlayedCubit>(
+      create: (context) => RecentlyPlayedCubit(),
+      child: BlocBuilder<RecentlyPlayedCubit,
+          ApiCallState<List<RecentlyPlayedSong>>>(
+        builder: (context, state) => state.map(
+          loading: (_) => const SizedBox(),
+          error: (_) => const SizedBox(),
+          data: (rpSongs) => BlocProvider<SongsCubit>(
+            create: (context) => SongsCubit(
+              rpSongs.data.map((e) => e.id).toList(),
+            ),
+            child: BlocBuilder<SongsCubit, ApiCallState<List<Song>>>(
+              builder: (context, state) => state.map(
+                loading: (_) => const SizedBox(),
+                error: (_) => const SizedBox(),
+                data: (songs) => SizedBox(
+                  child: ListView.builder(
+                    itemBuilder: (context, index) => SongRow(
+                      song: songs.data[index],
+                      isLikeAvailable: false,
+                    ),
+                    itemCount: songs.data.length,
+                    physics: const NeverScrollableScrollPhysics(),
+                  ),
```

Comment:
No need for this builder tree, change RecentlyPlayedCubit state to ApiCallState<List<Song>, load songs inside it and remove nested BlocProvider and BlocBuilder<SongsCubit, ...>